grid2op/tests/test_alert.py

```diff
@@ -13,45 +13,1234 @@
+            kwargs_opponent = dict(lines_attacked=[ATTACKED_LINE], 
+                                   duration=10, 
+                                   steps_attack=[0,10])
+            self.env = make(self.env_nm, test=True, difficulty="1", 
+                            opponent_attack_cooldown=0, 
+                            opponent_attack_duration=99999, 
+                            opponent_budget_per_ts=1000, 
+                            opponent_init_budget=10000., 
+                            opponent_action_class=PlayableAction, 
+                            opponent_class=TestOpponent, 
+                            kwargs_opponent=kwargs_opponent, 
+                            reward_class=AlertReward(reward_end_episode_bonus=42))
 
-        
         self.env.seed(0)
         self.env.reset()
-        assert isinstance(self.env.parameters.ALERT_TIME_WINDOW, np.int32)
-        assert self.env.parameters.ALERT_TIME_WINDOW > 0
+        self.do_nothing = self.env.action_space({})
 
-        param = self.env.parameters
-        param.init_from_dict({"ALERT_TIME_WINDOW": -1})
+    def tearDown(self) -> None:
+        self.env.close()
+
+    def test_init_default_param(self) -> None : 
+        env = make(self.env_nm, test=True, difficulty="1")
+        assert isinstance(env.parameters.ALERT_TIME_WINDOW, np.int32)
+        assert isinstance(env._attention_budget, LinearAttentionBudgetByLine)
+        assert env._opponent_class == GeometricOpponent
+        assert env.parameters.ALERT_TIME_WINDOW > 0
+
+        param = env.parameters
+        param.init_from_dict({
+            "ALERT_TIME_WINDOW": -1, 
+            "ASSISTANT_WARNING_TYPE": "BY_LINE"})
         
         negative_value_invalid = False
         try: 
-            self.env.change_parameters(param)
-            self.env.reset()
+            env.change_parameters(param)
+            env.reset()
         except : 
             negative_value_invalid = True 
 
         assert negative_value_invalid
 
+        # test observations for this env also
+        true_alertable_lines = ALL_ATTACKABLE_LINES
+        
+        assert isinstance(env.alertable_line_names, list)
+        assert sorted(env.alertable_line_names) == sorted(true_alertable_lines)
+        assert env.dim_alerts == len(true_alertable_lines)
 
 
+    def test_init_observation(self) -> None :
+        true_alertable_lines = [ATTACKED_LINE]
+        
+        assert isinstance(self.env.alertable_line_names, list)
+        assert sorted(self.env.alertable_line_names) == sorted(true_alertable_lines)
+        assert self.env.dim_alerts == len(true_alertable_lines)
+
+
+    def test_raise_alert_action(self) -> None :
+        attackable_line_id = 0
+        # raise alert on line number line_id
+        act = self.env.action_space()
+        act.raise_alert = [attackable_line_id]
+
+        act_2 = self.env.action_space({"raise_alert": [attackable_line_id]})
+        
+        assert act == act_2 
```

Comment:
did you also tested when multiple lines are "alerted" (so when we raise multiple alert on multiple lines)